weston-ivi-shell/src/ivi-controller.c

```diff
@@ -1820,6 +1833,9 @@ surface_event_create(struct wl_listener *listener, void *data)
            (struct ivi_layout_surface *) data;
     uint32_t id_surface = 0;
 
+    if (is_desktop_surface_create(shell, layout_surface))
```

Comment:
A much more readable condition would be:
if (is_desktop_surface() && is_surface_id_invalid())

Reply:
yeah, the condition should be: `if ((id_surface == IVI_INVALID_ID) && is_desktop_surface())`. it will avoid to check the is_desktop_surface() for the valid id.